script/WireAdapters.s.sol

```diff
@@ -78,41 +73,76 @@ contract WireAdapters is Script {
             string memory remoteConfig = fetchConfig(remoteNetwork);
             uint16 remoteCentrifugeId = uint16(vm.parseJsonUint(remoteConfig, "$.network.centrifugeId"));
 
-            // Register ALL adapters for this destination chain
+            // Build per-destination adapter set, excluding adapters not deployed on remote
+            IAdapter[] memory temp = new IAdapter[](3);
+            uint8 count = 0;
+
+            bool remoteWormhole = false;
+            bool remoteLayerZero = false;
+            bool remoteAxelar = false;
+
+            try vm.parseJsonBool(remoteConfig, "$.adapters.wormhole.deploy") returns (bool dep) {
+                remoteWormhole = dep;
+            } catch {}
+            try vm.parseJsonBool(remoteConfig, "$.adapters.layerZero.deploy") returns (bool dep) {
+                remoteLayerZero = dep;
+            } catch {}
+            try vm.parseJsonBool(remoteConfig, "$.adapters.axelar.deploy") returns (bool dep) {
+                remoteAxelar = dep;
+            } catch {}
```

Comment:
Maybe you can extract this into an utility (similar to the already exist on LaunchDeployer.sol)

```
    function _parseJsonBoolOrDefault(string memory config, string memory path) private pure returns (bool) {
        try vm.parseJsonBool(config, path) returns (bool value) {
            return value;
        } catch {
            return false;
        }
    }
``